crates/analyzer-host/src/tests/unit_tests.rs

```diff
@@ -0,0 +1,62 @@
+mod fsm_tests {
+	use std::sync::Arc;
+
+	use serde_json::Value;
+
+	use crate::{
+		fs::LspEnumerableFileSystem,
+		fsm::LspProtocolMachine,
+		json_rpc::message::{Message, Notification, Request},
+		lsp::{request::RequestManager, state::LspServerState},
+	};
+
+	#[test]
+	fn test_lsp_states() {
+		let rm = RequestManager::new(async_channel::unbounded::<Message>());
+		let mut lsp =
+			LspProtocolMachine::new(None, rm.clone(), Arc::new(Box::new(LspEnumerableFileSystem::new(rm.clone()))));
+		assert_eq!(lsp.current_state(), LspServerState::ActiveUninitialized);
+		assert_eq!(lsp.is_active(), true);
+
+		let mut params = serde_json::json!(analyzer_abstractions::lsp_types::InitializeParams { ..Default::default() });
+		let mut message =
```

Comment:
Default messages can be find in the `tester` cargo, and if the default message doesn't already exist could you add it:
https://github.com/p4lang/p4analyzer/blob/774e54da71c115d9c849ab3caa70ebd7fda9f865/crates/tester/src/tester.rs#LL45